src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -13,318 +13,332 @@ import IndexPatternsFieldListProvider from 'ui/index_patterns/_field_list';
+      const names = _.pluck(scriptedFields, 'name');
 
-      if (_.contains(scriptFields, name)) {
+      if (_.contains(names, name)) {
         throw new errors.DuplicateField(name);
       }
 
-      self.fields.push({
+      this.fields.push({
         name: name,
         script: script,
         type: type,
         scripted: true,
         lang: lang
       });
 
-      self.save();
-    };
+      this.save();
+    }
 
-    self.removeScriptedField = function (name) {
-      let fieldIndex = _.findIndex(self.fields, {
+    removeScriptedField(name) {
+      const fieldIndex = _.findIndex(this.fields, {
         name: name,
         scripted: true
       });
+      this.fields.splice(fieldIndex, 1);
+      this.save();
+    }
 
-      self.fields.splice(fieldIndex, 1);
-
-      self.save();
-    };
-
-    self.popularizeField = function (fieldName, unit) {
-      if (unit == null) unit = 1;
-
-      let field = _.get(self, ['fields', 'byName', fieldName]);
-      if (!field) return;
-
-      let count = Math.max((field.count || 0) + unit, 0);
-      if (field.count !== count) {
-        field.count = count;
-        self.save();
+    popularizeField(fieldName, unit = 1) {
+      const field = _.get(this, ['fields', 'byName', fieldName]);
+      if (!field) {
+        return;
       }
-    };
+      const count = Math.max((field.count || 0) + unit, 0);
+      if (field.count === count) {
+        return;
+      }
+      field.count = count;
+      this.save();
+    }
 
-    self.getNonScriptedFields = function () {
-      return _.where(self.fields, { scripted: false });
-    };
+    getNonScriptedFields() {
+      return _.where(this.fields, { scripted: false });
+    }
 
-    self.getScriptedFields = function () {
-      return _.where(self.fields, { scripted: true });
-    };
+    getScriptedFields() {
+      return _.where(this.fields, { scripted: true });
+    }
 
-    self.getInterval = function () {
+    getInterval() {
       return this.intervalName && _.find(intervals, { name: this.intervalName });
-    };
-
-    self.toIndexList = function (start, stop, sortDirection) {
-      return self
-      .toDetailedIndexList(start, stop, sortDirection)
-      .then(function (detailedIndices) {
-        if (!_.isArray(detailedIndices)) {
-          return detailedIndices.index;
-        }
-
-        return _.pluck(detailedIndices, 'index');
-      });
-    };
+    }
 
-    self.toDetailedIndexList = Promise.method(function (start, stop, sortDirection) {
-      let interval = self.getInterval();
+    toIndexList(start, stop, sortDirection) {
+      return this
+        .toDetailedIndexList(start, stop, sortDirection)
+        .then(detailedIndices => {
+          if (!_.isArray(detailedIndices)) {
+            return detailedIndices.index;
+          }
+          return _.pluck(detailedIndices, 'index');
+        });
+    }
 
+    toDetailedIndexList(start, stop, sortDirection) {
+      const interval = this.getInterval();
```

Comment:
I appreciate the desire to not use `Promise.method` here but this method still has the potential to throw synchronously. Maybe a high-level `Promise.try()` would be a bit more ergonomic and accomplish the same guarantee of returning a promise?